packages/runtime-core/src/renderEffect.ts

```diff
@@ -0,0 +1,28 @@
+import { ReactiveEffect } from '@vue/reactivity'
+
+/**
+ * Extend `ReactiveEffect` by adding `pause` and `resume` methods for controlling the execution of the `render` function.
+ */
+export class RenderEffect extends ReactiveEffect {
+  private _isPaused = false
+  private _isCalled = false
+  pause() {
+    this._isPaused = true
+  }
+  resume(runOnce = false) {
+    if (this._isPaused) {
+      this._isPaused = false
+      if (this._isCalled && runOnce) {
+        super.run()
+      }
+      this._isCalled = false
+    }
+  }
+  update() {
+    if (this._isPaused) {
+      this._isCalled = true
+    } else {
+      return super.run()
+    }
+  }
```

Comment:
Love the compatibility of pausable effect. Since this is not bound to the runtime/dom, I wonder if this could be generally available in `ReactiveEffect` directly.

/cc @yyx990803 WDYT? Or do you think it deserves a dedicated RFC?

Reply:
As the addition of `pause` and `resume` methods in `ReactiveEffect` has significant implications for the existing codebase, I have initiated an RFC to discuss this feature.

https://github.com/vuejs/rfcs/discussions/599